ThunderRequest/BackgroundSessionController.swift

```diff
@@ -56,7 +62,11 @@ public class BackgroundRequestController: NSObject, URLSessionDelegate, URLSessi
             responseHandler?(downloadTask, nil, nil)
             return
         }
-        let response = RequestResponse(response: taskResponse, data: try? Data(contentsOf: location))
+        var data: Data?
+        if readData {
```

Comment:
Good idea to save a (possibly) unnecessary read :).

Makes sense why `URLSessionDownloadTask` only returns a file `URL.
Then again surely `URLSessionDownloadTask` is more or less the same as an `URLSessionDataTask` just saving to a file on successful response. So a `URLSessionDataTask` then writing that data manually in the callback is the same time overall?

Reply:
So I think `URLSessionDataTask` would be incrementally saving to disk on download (Certainly with the app in the background) so unsure of that, but just trying to do everything we can from our side to reduce the memory footprint as there is a much discussed (But not documented) 40mb seemingly with `urlsessiond` which handles background downloads :D 

`URLSessionDownloadTask` is required in this particular case because `URLSessionDataTask` won't execute in the background :)